src/compiler/binder.ts

```diff
@@ -2304,7 +2324,10 @@ namespace ts {
                 return s;
             });
             if (symbol) {
-                declareSymbol(symbol.exports, symbol, lhs, SymbolFlags.Property | SymbolFlags.ExportValue, SymbolFlags.None);
+                const flags = isClassExpression(node.right) ?

```

Comment:
Now that `typedef`s are exported, there's a possibility that their names collide, such that there is a difference between

```ts
let x: import("./a.js").Foo
// and
import a = require('./a.js')
var foo = new Foo()
```

In other words, the type reference `Foo` might refer to the typedef, whereas the value referenced in `new Foo` might refer to the class. There should be an error to prevent this.

Reply:
i see. thanks.